app/controllers/api/v1/graphql_controller.rb

```diff
@@ -19,7 +23,86 @@ def query
           params[:query],
           :variables => variables
         )
-        render :json => result
+        meta = calculate_metadata(result) || {}
+        apply_limits_and_offsets(result)
+
+        render :json => result.to_h.merge({'meta' => meta})
+      ensure
+        Api::V1::GraphqlController.limits_and_offsets = {}
+      end
+
+      private
+
+      def calculate_metadata(data)
+        result = {}
+
+        data.to_h["data"]&.each do |o|
+          result[o.first] = o.second.size
+        end
+
+        {"count" => result}
+      end
+
+      def apply_limits_and_offsets(data)
+        result = data.to_h["data"]
+        limits_and_offsets = Api::V1::GraphqlController.limits_and_offsets
+        return unless limits_and_offsets
+
+        result&.each do |o|
+          name = o.first
+          args = limits_and_offsets[name]
+
+          result[name] = apply_offset(result[name], args["offset"]) if args["offset"]
+          result[name] = apply_limit(result[name], args["limit"]) if args["limit"]
+        end
+      end
+
+      def apply_offset(data, offset)
+        offset <= data.size ? data[offset..-1] : []
+      end
+
+      def apply_limit(data, limit)
+        data[0..limit - 1]
+      end
+    end
+  end
+end
+
+module GraphQL
+  module Execution
+    class Execute
+      class << self
+        def begin_query(query, _multiplex)
+          remove_offsets_and_limits(query)
```

Comment:
I think it can't work this way, removing limit from request can cause returning milions of records. SQL count should be used instead